modules/puppetca/puppetca_api.rb

```diff
@@ -2,14 +2,18 @@
 
 module Proxy::PuppetCa
   class Api < ::Sinatra::Base
+    extend Proxy::PuppetCa::DependencyInjection
+    inject_attr :cert_manager, :cert_manager
```

Comment:
I'd call this `ca` or `ca_interface` or something similar.

E.g. `inject_attr :ca_provider, :ca` reads a lot cleaner in my opinion.

Reply:
Well `ca` is the whole thing, which consists of managing certificates and autosign-entries. Since the `cert_manager` basically uses the `puppet cert` command for everything it does, I thought this would be matching name.

Reply:
I don't see autosigning as part of the CA. Basically you have two separate strings:

1. Interface to the CA (ask the ca to issue/remove/... a certificate, ...)
2. Interface to an autosigning provider (list whitelisted hosts, whitelist a host, ...). This is typically not part of the CA, but of the RA.

Certmanager is definitely misleading. Please use something else.

Reply:
Well, its part of the puppetCA module and under the `puppet/ca/` api endpoint...
But I can understand that that may be confusing, how about just `puppet_cert` so its obvious it uses the `puppet cert` command?

Reply:
Ok, works for me.